jupyterhub/templates/hub/rbac.yaml

```diff
@@ -19,6 +19,14 @@ rules:
   - apiGroups: [""]       # "" indicates the core API group
     resources: ["events"]
     verbs: ["get", "watch", "list"]
+  {{- if .Values.proxy.externalIngressController }}
+  - apiGroups: [""]       # "" indicates the core API group
+    resources: ["endpoints", "services"]
+    verbs: ["get", "watch", "list", "create", "delete", "patch"]
```

Comment:
To help with review did you figure these out by going through the K8s API calls, or is it a guess?

Reply:
Additive hardworking way. I added permission according to hub error logs.